codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -272,6 +275,12 @@ private boolean readClientPrefaceString(ByteBuf in) throws Http2Exception {
             if (bytesRead == 0 || !ByteBufUtil.equals(in, in.readerIndex(),
                                                       clientPrefaceString, clientPrefaceString.readerIndex(),
                                                       bytesRead)) {
+                String http1Request = in.toString(0, 1024, CharsetUtil.US_ASCII);
+                int http1Index;
+                if ((http1Index = http1Request.indexOf("HTTP/1.")) != -1) {
+                    throw connectionError(PROTOCOL_ERROR, "Unexpected HTTP/1 request: %s",
```

Comment:
This is careful to not include any of the header data (which would include cookies and auth tokens and such).  It should limit to the first line, "GET /secretGovernmentFiles HTTP/1.1\r\n".  The hexdump does reveal some of it too (albeit less).